src/distilabel/llm/mistralai.py

```diff
@@ -0,0 +1,157 @@
+
+        self.max_tokens = max_tokens
+        self.temperature = temperature
+        self.top_p = top_p
+        self.seed = seed
+        self.safe_prompt = safe_prompt
+
+        # If the api key is not given, try to get it from the environment
+        if not api_key:
+            api_key = os.environ.get("MISTRALAI_API_KEY")
+        # Explicitly write the default parameters of the model
+        self.client = client or MistralClient(
+            api_key=api_key, max_retries=5, timeout=120
+        )
+        assert (
+            model in self.available_models
+        ), f"Provided `model` is not available in MistralAI, available models are {self.available_models}"
+        self.model = model
+
+    def __rich_repr__(self) -> Generator[Any, None, None]:
+        yield from super().__rich_repr__()
+        yield (
+            "parameters",
+            {
+                "max_tokens": self.max_tokens,
+                "temperature": self.temperature,
+                "top_p": self.top_p,
+                "seed": self.seed,
+                "safe_prompt": self.safe_prompt,
+            },
+        )
+
+    @cached_property
+    def available_models(self) -> List[str]:
+        """Returns the list of available models in MistralAI."""
+        return [model.id for model in self.client.list_models().data]
+
+    @property
+    def model_name(self) -> str:
+        """Returns the name of the MistralAI model."""
+        return self.model
+
+    def _generate(
+        self,
+        inputs: List[Dict[str, Any]],
+        num_generations: int = 1,
+    ) -> List[List[LLMOutput]]:
+        """Generates `num_generations` for each input in `inputs`.
+
+        Args:
+            inputs (List[Dict[str, Any]]): the inputs to be used for generation.
+            num_generations (int, optional): the number of generations to be performed for each
+                input. Defaults to 1.
+
+        Returns:
+            List[List[LLMOutput]]: the generated outputs.
+        """
+        prompts = self._generate_prompts(inputs, default_format="openai")
+        # The mistralai format is the same as openai, but needs to be converted to mistralai's ChatMessage (pydantic model)
+        prompts = [[ChatMessage(**p) for p in prompt] for prompt in prompts]
+        outputs = []
+        for prompt in prompts:
+            responses = []
+            for _ in range(num_generations):
+                chat_completion_response = self.client.chat(
+                    self.model,
+                    messages=prompt,
+                    temperature=self.temperature,
+                    max_tokens=self.max_tokens,
+                    top_p=self.top_p,
+                    seed=self.seed,
+                    safe_prompt=self.safe_prompt,
+                )
+                responses.append(chat_completion_response)
+
+            output = []
+            for response in responses:
+                chat_completion = response.choices[0]
+                try:
```

Comment:
I believe this might be missing in some other implementations?

Reply:
will take a look, but maybe this is a special behaviour of this API, it doesn't deal with `num_generations` in the same way as OpenAI does, so it's easier to grab that choice